retis/src/core/probe/kernel/probe_stack.rs

```diff
@@ -44,11 +38,25 @@ impl ProbeStack {
 
         Self {
             probes: HashSet::from_iter(attached_probes),
-            keep_stack,
             known_kernel_types,
         }
     }
 
+    fn keep_stack(&mut self, mgr: &mut ProbeRuntimeManager, evt: &KernelEvent) -> bool {
+        let r#type = match evt.probe_type.as_str() {
+            "raw_tracepoint" => "tp",
+            s => s,
+        };
+
+        let sym = format!("{}:{}", r#type, evt.symbol);
+
+        if let Some(opts) = mgr.get_probe_opts(&sym) {
+            opts.contains(&crate::core::probe::ProbeOption::ReportStack)
+        } else {
+            false
+        }
```

Comment:
nit (since there were other comments): I think this can be simplified to:

```rust
mgr.get_probe_opts(&sym).is_some_and(|opts| opts.contains(&crate::core::probe::ProbeOption::ReportStack))
```

(Super nit: you might want to import `ProbeOption` instead of using the fully qualified name).